exporter/exporterhelper/internal/queue/persistent_queue.go

```diff
@@ -350,10 +420,8 @@ func (pq *persistentQueue[T]) onDone(index uint64, itemsSize int64, bytesSize in
 		pq.metadata.ItemsSize = 0
 	}
 
-	// Ensure the used size are in sync when queue is drained.
-	if pq.requestSize() == 0 {
-		pq.metadata.BytesSize = 0
-		pq.metadata.ItemsSize = 0
+	if err := pq.itemDispatchingFinish(context.Background(), index); err != nil {
+		pq.logger.Error("Error deleting item from queue", zap.Error(err))
```

Comment:
Why is this function (onDone) needs to be changed?

Reply:
The `pq.requestSize() == 0` logic was moved to `itemDispatchingFinish` to avoid duplicate code.